### PR TITLE
update and delete agenda item and mark task as Done

### PR DESCRIPTION
**Pull request by Gerondk**

### CODE REVIEW

File: agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt [hunk 1]
```diff
@@ -51,6 +50,55 @@ class OfflineAgendaRepository(
...
+                    val eventBody = agendaItem.toEventBody()
+                    val eventBodyPart = buildEventBodyPart(eventBody)
+                    val photosPart = buildPhotosPart(agendaItem.photos)
+                    taskyRetrofitApi.updateEvent(
+                        eventBodyPart = eventBodyPart,
+                        photosPart = photosPart
+                    )
+                }
+
+            }
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt:66]: Could it also make sense to update your local database here?

File: agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt [hunk 2]
```diff
@@ -51,6 +50,55 @@ class OfflineAgendaRepository(
...
+        }.launchIn(scope)
+    }
+
+    override fun deleteAgendaItem(agendaItem: AgendaItem) {
+        when(agendaItem){
+            is AgendaItem.Event -> {
+                networkApiCall {
+                    taskyRetrofitApi.deleteEvent(agendaItem.id)
+                }
+            }
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt:86]: Also, here you delete the agenda item on the remote server, but your app will still have it locally

File: agenda/presentation/src/main/java/com/gkp/agenda/presentation/edit/util/Util.kt [hunk 3]
```diff
@@ -58,6 +59,39 @@ fun getReminderInDateLong(
...
+    if (substractDate == startDate) {
+      return ReminderTimes.TEN_MINUTES_BEFORE.textId
+    }
+    substractDate = reminderDate.plusMinutes(30)
+    if (substractDate == startDate) {
+       return ReminderTimes.THIRTY_MINUTES_BEFORE.textId
+    }
+    substractDate = reminderDate.plusHours(1)
+    if (substractDate == startDate) {
+      return ReminderTimes.ONE_HOUR_BEFORE.textId
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/gkp/agenda/presentation/edit/util/Util.kt]: You can simplify this code

```
fun reminderLongToReminderTimeTextId(reminderLong: Long, startLong: Long): Int {
    val reminderDate = LocalDateTime.ofInstant(
        java.time.Instant.ofEpochMilli(reminderLong),
        ZoneId.systemDefault()
    )
    val startDate = LocalDateTime.ofInstant(
        java.time.Instant.ofEpochMilli(startLong),
        ZoneId.systemDefault()
    )

    return when {
        reminderDate.plusMinutes(10) == startDate -> ReminderTimes.TEN_MINUTES_BEFORE.textId
        reminderDate.plusMinutes(30) == startDate -> ReminderTimes.THIRTY_MINUTES_BEFORE.textId
        reminderDate.plusHours(1) == startDate -> ReminderTimes.ONE_HOUR_BEFORE.textId
        reminderDate.plusHours(6) == startDate -> ReminderTimes.SIX_HOURS_BEFORE.textId
        reminderDate.plusDays(1) == startDate -> ReminderTimes.ONE_DAY_BEFORE.textId
        else -> 0
    }
}
```

**Review by Kai-Cilliers — APPROVED** [review]

File: agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt (hunk 1 above)
Comment (Gerondk) [reply agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt:66]: I will implement the caching in the next PR

**Pull request merged**